juju/juju
# acceptancetests cmddir: additional cli test for local charm deploy

## Description of change

Adds an acceptance test to check whether the correct working directory will be used for a deploy of a local charm 

## QA steps
- depends on this PR to be merged first: https://github.com/juju/charm/pull/293
- run the testsuite: `./main.sh cli test_cwd_no_git `

e.g. output of success
```
[15:51:07] nam:tests git:(add-test-cwd*) $ ./main.sh cli test_local_charm -v

==> Checking for dependencies
==> TEST BEGIN: test_cli
==> Checking for dependencies
====> Bootstrapping juju (2.7)
====> Bootstrapped juju (1s)
====> Reusing bootstrapped juju (2.7)
Added 'local-charm-deploy' model on localhost/localhost with credential 'localhost' for user 'admin'
====> Bootstrapped juju (1s)
cloning ntp charm
Success: "exit status 128" not found
====> Destroying juju model local-charm-deploy
====> Destroyed juju model local-charm-deploy
====> Reusing bootstrapped juju (2.7)
Added 'local-charm-deploy-wrong-git' model on localhost/localhost with credential 'localhost' for user 'admin'
====> Bootstrapped juju (1s)
cloning ntp charm
creating local git folder
Initialized empty Git repository in /tmp/ci-wMXdPsNSgy/.git/
[master (root-commit) 5337a25] rand_file
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 rand_file
Deploying charm "local:bionic/ntp-0".
"43f8719"
"43f8719"
Juju status returns the expected sha 
====> Destroying juju model local-charm-deploy-wrong-git
====> Destroyed juju model local-charm-deploy-wrong-git
==> TEST DONE: test_cli (7s)
==> Cleaning up
====> Cleaning up jujus

==> Test result: success
==> Tests Removed: /home/nam/golang/src/github.com/juju/juju/tests/tmp.5xF
==> TEST COMPLETE
```
## Documentation changes
nothing
## Bug reference
https://bugs.launchpad.net/juju/+bug/1789447
